### DISCUSSION TITLE
vs code formatting mixed php and html

### DISCUSSION THREAD

**Discussion by justinblayney** [squizlabs/PHP_CodeSniffer#3941] [Q&A]
I have set up formatting in vs code, and it works great for css and php when they are by themselves, the problem is when I mix PHP and HTML.

for example I want

```
<div>
   <?php>
      <?php>
         <div></div>
      <?php>
   <?ph>
</div>

```
on save, it will make it

```
<div>
<?php>
<?php>
   <div></div>
<?php>
<?php>
</div>

```
I know I shouldn't mix the 2, but the PHP is just loops and conditional statements, so much easier to write as I am.

Any way to do this?

**Comment by fredden** [comment]
You can run `phpcs` with the `-s` flag to show the sniffs which are complaining, then you can disable those sniffs in your custom standard or per-file.

References:
- https://github.com/squizlabs/PHP_CodeSniffer/wiki/Reporting#printing-full-and-summary-reports
- https://github.com/squizlabs/PHP_CodeSniffer/wiki/Advanced-Usage#ignoring-parts-of-a-file
- https://github.com/squizlabs/PHP_CodeSniffer/wiki/Annotated-Ruleset

**Reply by justinblayney** [reply]
Hello,
is there a way to run it only on the page im working on, that command grabs everything, including vendor folders which gets 1000s of hits and isn't useful

**Reply by jrfnl** [reply]
Presuming you are using git: `phpcs -s --filter=GitModified`

**Comment by lukajincharadze** [comment]
Install PHP Extension: Ensure you have the PHP extension installed in VS Code. This extension provides PHP language support and formatting features.

Configure Formatting Settings:

Open your VS Code settings (press Ctrl + , or go to File > Preferences > Settings).
Search for "format" in the search bar to filter formatting-related settings.
Adjust the settings for PHP and HTML formatting to your preference. For example:
`"php.format.enable": true,
"php.suggest.basic": false,
"editor.formatOnSave": true, // Enable format on save
"editor.defaultFormatter": "esbenp.prettier-vscode", // Use Prettier as the default formatter
`

You can also specify specific settings for HTML formatting within PHP files::

`"[php]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode"
}
`


Use Prettier or Other Formatter: Consider using a code formatter like Prettier for consistent formatting across HTML and PHP code blocks. Prettier supports formatting both PHP and HTML code, ensuring uniform formatting styles.

Ensure Proper Indentation: Make sure your PHP and HTML code blocks are properly indented to maintain readability. Proper indentation helps differentiate between PHP and HTML blocks and improves code organization.

**Reply by jrfnl** [reply]
AI bot detected...

**Reply by devtz007** [reply]
Prettier doesn't support php.

**Reply by simonLeary42** [reply]
> Prettier doesn't support php.

https://github.com/prettier/plugin-php

**Comment by pixelmultiplo** [comment]
Is there a way to do this? It's pretty annoying.
